### PR TITLE
LCov Coverage Checks within GitHub Actions

### PR DESCRIPTION
**Pull request by kzscisoft**
Execution of the code coverage summary tool `lcov` (which runs `gcov`) has now been added to GitHub actions. A summary PDF file depicting coverage for each source script is generated within the artifacts for the given workflow.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,11 +5,58 @@ set(PROJECT_NAME Covid19EERAModel)
...
+set(COVERAGE_COMPILER_FLAGS "-g -fprofile-arcs -ftest-coverage")
+set (CMAKE_CXX_FLAGS "-std=gnu++11 -Wall -Wextra -O0 ${COVERAGE_COMPILER_FLAGS} -DROOT_DIR=\\\"${CMAKE_SOURCE_DIR}\\\"")
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 add_subdirectory(src)
 add_subdirectory(test)
+
+find_program( LCOV_PATH  NAMES lcov lcov.bat lcov.exe lcov.perl)
```
Comment (peter-t-fox) [inline CMakeLists.txt:20]: Is it the case that people who do not have these things installed still be able to run the build on their machines, or will this cause their builds to error out?

Comment (kzscisoft) [reply CMakeLists.txt:20]: If it fails to find `lcov` then the variable is not set and then `make coverage` would fail. The main build should not fail.

Comment (qingfengxia) [reply CMakeLists.txt:20]: I quite like the workflow yml,  for the CMakeLists.txt, I hope for a little tweak

```
if (NOT DEFINED CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Debug)
endif()
```
In the short term, we default to debug, later to Release. Currently, user can overwrite this by  "cmake .. -DCMAKE_BUILD_TYPE=Release" in benchmarking. 

For coverage, we may make an cmake option, default to OFF.    So the user will not even notice there is coverage stuff.     

While this is only my way of doing things,  for your reference.
https://github.com/qingfengxia/cpp_numeric_cast/blob/master/CMakeLists.txt

**Comment by kzscisoft** [comment]
@qingfengxia Great ideas, I have added them to: #21

**Comment by qingfengxia** [comment]
Sorry, again, I always comment on outdated things. sorry

________________________________

**Pull request merged**